docs/examples/asynchronous_consumer_example.rst

```diff
@@ -334,7 +334,7 @@ consumer.py::
             LOGGER.info('Stopping')
             self._closing = True
             self.stop_consuming()
-            self._connection.ioloop.start()
+            self._connection.ioloop.stop()
```

Comment:
I think there is more to it than this. `stop_consuming` sends a message to the broker and expects a response callback. If you stop the ioloop now, the response won't arrive and the response callback won't be handled. More needs to be done to make these states work correctly. See https://github.com/pika/pika/pull/710/files for an example of cleanup in the Async Publisher code, which is quite similar to the consumer code.